### ISSUE TITLE
Multiple `transfer-encoding: chunked` requests completely breaks NextJS

### ISSUE THREAD

**Issue by ShanonJackson** [vercel/next.js#15721]
# Bug reportz
// {target: "server"} for context.

It seems like there is some "blocking" behavior when a request returns a chunked response, this blocking behaviour is breaking our recently built/deployed production app by basically capping concurrent requests at around 6 (at the server level).

## Describe the bug
```
export default (req, res) =>{
    res.write("\n");
    req.on("close", () => {
	 res.end();
    });
}
write this endpoint say `/api/stream` and open 6-7 browser tabs then try refresh the homepage on another tab and watch the server completly cap out; all requests will be stuck on "pending" definitely.

Our real world use case actually streams data obviously however this is a minimally reproducible example.
```

TL;DR call this endpoint on 6 tabs, and just refresh homepage on 7th will just be stuck on pending indefinitely as its 'blocking' at the server level

## Expected behavior

Server can handle factors of 100 more concurrent requests; potentially up to thousands.

## System information

- OS: Windows
- Browser Chrome
- Version of Next.js: 9.4.4
- Version of Node.js: 10.16.3

**Comment by ShanonJackson** [comment]
Update Fuck my life i think its accidentally "blocking" in NodeJS when it shouldn't be?

```
const express = require('express')
const app = express()
const port = 3000

app.get('/', (req, res) => res.send('Hello World!'));
app.get("/stream", (req,res) => {
	res.write("HELLO WORLD\n");
	req.on("close", () => {
		res.end();
	})
});

app.listen(port, () => console.log(`Example app listening at http://localhost:${port}`))
```

Same thing, open 6 tabs on /stream then open a 7th on homepage (localhost:3000) and the homepage will never load unless you start closing the /stream tabs.

So this bug also exists within express.

What can we do about this NextJS team?

**Comment by ShanonJackson** [comment]
No worrys; got to bottom of it. Chrome will only let you have 6 concurrent active connections to 1 host so its not blocking in the server level. Sorry for the stress

**Comment by Timer** [comment]
No problem!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
